lib/miq_automation_engine/engine/miq_ae_service/miq_ae_service_rbac.rb

```diff
@@ -0,0 +1,37 @@
+module MiqAeMethodService
+  module MiqAeServiceRbac
+    extend ActiveSupport::Concern
+
+    module ClassMethods
+      def find_ar_object_by_id(id)
+        MiqAeService.set_current_user if MiqAeService.rbac_enabled?
+        MiqAeService.rbac_enabled? ? Rbac.search(:class => model, :targets => [id], :results_format => :objects).first.first : model.send(:find, *id)
```

Comment:
Do not change:

1. use `Rbac.filtered(model)` where you can instead of `Rbac.search`
2. Use `model.find(*id)` instead of `send`